site/src/examples/combo-box/ServerSideData.tsx

```diff
@@ -42,25 +49,42 @@ export const ServerSideData = (): ReactElement => {
     }
   };
 
+  const hasResults = Array.isArray(data) && data.length > 0;
+
   return (
-    <ComboBox
-      onChange={handleChange}
-      onSelectionChange={handleSelectionChange}
-      value={value}
-      style={{ width: "266px" }}
-      endAdornment={loading && <Spinner size="small" />}
-    >
-      {!loading ? (
-        data?.map((color) => <Option value={color} key={color} />)
-      ) : (
-        <div
-          className={styles.statusOption}
-          role="option"
-          aria-selected="false"
-        >
-          Loading...
-        </div>
-      )}
-    </ComboBox>
+    <FormField style={{ width: "266px" }}>
+      <FormFieldLabel>State</FormFieldLabel>
+      <ComboBox
+        onChange={handleChange}
+        onSelectionChange={handleSelectionChange}
+        value={value}
+        endAdornment={loading && <Spinner size="small" />}
+      >
+        {!loading && hasResults
+          ? data?.map((state) => <Option value={state} key={state} />)
+          : null}
+        {!loading && !hasResults ? (
+          <div
+            className={styles.statusOption}
+            role="option"
+            aria-selected="false"
+          >
+            No results found
```

Comment:
Thinking about this around "min 2 char" message, should we show something different than no result when user first started typing?

Reply:
Ah, I thought I handled that case :(